doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
 
+### Argument for Correct read-back
+
+We wish to show that a read-back of an EAL-expression that has undergone zero or more reductions gives back the same result as normal evaluation up to alpha equality.
+
+Before we can prove this result, we must first prove a few lemmas and theorems first as well talk about what fan in's with the same label imply.
+
+The first lemma we wish to prove is the following
+
+\begin{Lemma}{AST→Net has one free port}{AST->Net}
+  Let $A$ = be a valid BOHM ADT.
+  \\ \\
+  Now Consider the net over $A$, let $N_a$ be this net.
+  \\ \\
+  Now let us consider any sub case of $A$, $L = P(n_1 \star \cdots \star n_k)$.
+  \\ \\
+  The node corresponding to $L$ looks.
+
+  \begin{tikzpicture}
+    \draw (0,0) circle (.8cm);
+  \end{tikzpicture}
+
+  Where the labels in $N_a$ are the same as in $L$, with $n_{k+1}$ being an extra port which connects to the ADT above it.
+  \\ \\
+  For an ADT to be realized, we must have all arguments, so all these ports must be connected.
+  Now, if $L$ is $A$, then we have one free port, however if it is not, then this free port must not be free but instead be connected to the ADT above it.
+  \\ \\
+  For free variables which have no $\lambda$ to connect to, a symbol node is created, maintaining the invariant.
+  \\ \\
+  $\therefore$ AST→Net has one free port
+\end{Lemma}
+
+
+Now we need to prove that evaluating this net does not change this fact
+
+
+\begin{Theorem}{AST→Net→Eval has one free port}{AST->Net->Eval}
+  Let $A = Net(AST)$.
+  \\ \\
+  By lemma \ref{Lm:AST->Net} $A$ has one free port.
+  \\ \\
+  We must now show $Eval(A)$ does not change the number of free ports.
+  \\ \\
+  We can show this by simply considering how every reduction rule works. For the sake of brevity, we will only consider the three variations of rewrite rules present in the BOHM system WLOG.
+
+  \begin{tcolorbox}[title=Case 1 - Rewire, colframe=inner-box-1]
+    \begin{figure}[H]
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \inetcell[]{a}{$g$}[5]
+          \inetcell[rotate = 180, below = 50 pt]{l}{\rotatebox[origin=c]{180}{$f$}}[5]
+          \inwire[]{a/1}{a}
+          \inwire[]{a/4}{d}
+          \outwire[]{l/1}{c}
+          \outwire[]{l/4}{b}
+          \cutWire{a.out}{l.out}{}{}
+        \end{tikzpicture}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.05\textwidth}
+        \huge{→}
+      \end{subfigure}
+      ~
+      \begin{subfigure}[c]{0.1\textwidth}
+        \begin{tikzpicture}
+          \wirecross[]{a}{c}{b}{d}
+        \end{tikzpicture}
+      \end{subfigure}
+    \end{figure}
+
+    We can see that there is only one wire eliminated, the one connecting the main ports of node $f$ and node $g$.
+    \\ \\
+    The rest of the wires are simply rewired to each other, thus not effecting the total number of free ports
+  \end{tcolorbox}
+
+  \begin{tcolorbox}[title=Case 2 - Isolation, colframe=inner-box-1]
```

Comment:
"Isolation"?

Reply:
It isolates the node from the rest of the graph (takes it out of the connected components), and connects an eraser to it